### ISSUE TITLE
Feature: Shared State with react native and react

### ISSUE THREAD

**Issue by codecomeon** [gronxb/webview-bridge#17]
for example. App inject the token, userInfo, business id, business info, web get the state; and web can change it, app will auto useEffect the change of any state.

It's a common usecase for every webview hybrid app.

**Comment by gronxb** [comment]
Thank you for your suggestion!!!

 I am currently developing the same feature you mentioned. 
My goal is to implement two-way binding between the web and React Native.

Of course, for the web part, I am considering a reactive state that does not depend on specific frameworks like React, Vue, or Svelte.

It's likely that libraries will be developed for integrating state across different frameworks as well. Examples could include `@webview-bridge/react`, `@webview-bridge/vue`.

**Comment by gronxb** [comment]
I'm working on a design, it's not finished, but this is probably how I'll use it.

## React Native Part
```tsx
// here
export const appStore = storeBridge({
  count: 0,
});

// here
export type AppStore = typeof appStore;

export const { store } = createWebView({
  // register store bridge  
  bridge: bind([appMethod, appStore]),

  // ... default options
});

const count = useSharedStore(store, (state) => state.count);
const setStore = useSetSharedStore(store);
setStore("count" , (prev) => prev + 1);
```

## Web
```tsx
import { linkNativeStore } from "@webview-bridge/react"; // or "@webview-bridge/vue"

const nativeStore = linkSharedStore<AppStore>();

const count = useSharedStore(store, (state) => state.count);
const setStore = useSetSharedStore(store);
setStore("count" , (prev) => prev + 1);
```

**Comment by gronxb** [comment]
shared state was released in v1.3.1 !!!!

Released a library to integrate with state in vanilla, react, and vue.
https://www.npmjs.com/package/@webview-bridge/react
https://www.npmjs.com/package/@webview-bridge/vue

https://gronxb.github.io/webview-bridge/shared-state/react-native.html
